src/components/edit/exchange-update/edit-exchange.js

```diff
@@ -0,0 +1,148 @@
+import i18n from '@dhis2/d2-i18n'
+import { Box, NoticeBox, ReactFinalForm } from '@dhis2/ui'
+import classNames from 'classnames'
+import PropTypes from 'prop-types'
+import React, { useCallback } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { AttributeProvider, useAppContext } from '../../../context/index.js'
+import { Loader } from '../../common/index.js'
+import { RequestEdit } from '../request-update/index.js'
+import { EditItemFooter, EditTitle } from '../shared/index.js'
+import { EditExchangeFormContents } from './edit-exchange-form.js'
+import styles from './edit-exchange.module.css'
+import { getInitialValuesFromExchange } from './getExchangeValues.js'
+import { useRequests } from './useRequests.js'
+import { useUpdateExchange } from './useUpdateExchange.js'
+
+const { Form } = ReactFinalForm
+
+const formatError = (error) => {
+    if (error.details?.response?.errorReports?.length > 0) {
+        return error.details.response.errorReports.reduce(
+            (stringified, rep) => {
+                if (rep.message) {
+                    return stringified + `\n${rep.message}`
+                }
+                return stringified
+            },
+            ''
+        )
+    }
+    return error?.message
+}
+
+export const EditExchange = ({ exchangeInfo, addMode }) => {
```

Comment:
This name is misleading as it's also used by the add-view, maybe `ExchangeForm` or something along those lines is better.

Reply:
Updated this name and other similar names, now using ExchangeForm, ExchangeFormContents, RequestForm, RequestFormContents